CMakeLists.txt

```diff
@@ -292,6 +295,18 @@ if(WIN32)
 else()
     # Custom build flags.
 
+    if (NOT CMAKE_SYSTEM_PROCESSOR STREQUAL arm)
+        set(MARCH -march=native)
```

Comment:
I definitely do not want this to be the default. We want to be able to distribute binaries that work on older CPUs. We don't control what CPU level is used by CI.

I'm not opposed to adding a way to set this with a flag, as I definitely understand what its there for.

Reply:
Should I just remove `-march=native` from the PR? Or maybe make it conditional on some flag (`QUIC_OPTIMIZE_MAX` or something like that)?

Reply:
I think its a good idea to have it, so a conditional flag is fine. Its just not a good deal.